### PR TITLE
Migrate or remove old repository unit tests

### PR DESCRIPTION
**Pull request by MatthewHambley**
Salvage some value from the technology demonstrator implementation, unit test wise. Also some minor type hinting, documentation and trying to work out what any of this stuff does. This change only covers repository access unit tests.

### CODE REVIEW

**Review by ericaneininger — COMMENTED** [review]
This all looks sensible to me.  Since the rest of the change is making a point of adding type hinting, there are a couple of places I think more could be added.

File: tests/unit_tests/steps/grab/test_git.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        git_checkout(config, parent_url)
+        assert (tmp_path / 'baz' / 'source' / '.git').exists()
+        file_tree_compare(workspace.tree_path, tmp_path / 'baz' / 'source')
+
+        process.send_signal(SIGTERM)
+        process.wait(timeout=2)
+        assert process.returncode == -15
+
+    @mark.skip(reason="Too hard to test at the moment.")
+    def test_extract_from_http(self, workspace, tmp_path: Path):
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply tests/unit_tests/steps/grab/test_git.py:109]: Yes

File: tests/unit_tests/steps/grab/test_git.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        Tests that an error is returned if the repository is not there.
+        """
+        config = BuildConfig(fab_workspace=tmp_path, project_label='bar')
+        parent_url = f'file://{tmp_path}/nosuch.repo'
+        with raises(FabException):
+            git_checkout(config, parent_url)
+
+    @mark.skipif(not find_executable('git-daemon'),
+                 reason="Unable to find git daemon")
+    def test_extract_from_git(self, workspace, tmp_path: Path):
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply tests/unit_tests/steps/grab/test_git.py:88]: Also yes.

File: tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py [hunk 3]
```diff
@@ -3,28 +3,106 @@
...
 
-from fab.steps.grab.svn import _get_revision
-import pytest
+from fab.build_config import BuildConfig
+from fab.steps.grab.svn import split_repo_url, svn_export
+
+from .support import Workspace, file_tree_compare
 
 
 class TestRevision(object):
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py:20]: Removed.

File: tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py [hunk 4]
```diff
@@ -3,28 +3,106 @@
 #  For further details please refer to the file COPYRIGHT
 #  which you should have received as part of this distribution
 # ##############################################################################
+from pathlib import Path
+from subprocess import Popen, run
+import time
+from typing import Tuple
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply tests/unit_tests/steps/grab/test_svn_fcm_unit_test.py]: Fixed.

**Comment by MatthewHambley** [comment]
This chnage has become totally unwieldy with all the style fixes. I'm shuttting it down so the repository migration can happen and I'll start again from scratch with separate changes.

**Pull request closed without merging**